src/actions/loadData.js

```diff
@@ -283,47 +283,78 @@ Dataset.prototype.fetchMain = function fetchMain() {
     .then((res) => res.json());
 };
 Dataset.prototype.fetchSidecars = async function fetchSidecars() {
+  /**
+   * If deemed appropriate, fetch sidecars and store the resulting promise as `this.<sidecarName>`.
+   * The returned promise is used by `loadSidecars` which will append `.then` clauses to the promise;
+   * often the load function is called immediately (and so the promises are unresolved) but for
+   * narratives it may be called some time later. Because of this the promises should not reject -
+   * if there is a fetch error then we resolve to the error and it is the responsibility of
+   * `loadSidecars` to react appropriately.
```

Comment:
_non-blocking_

It's not clear to me from anything in this commit why the rejected promises have to be converted to promises that resolve to an `Error`. Why can't the rejection be chained thru? It's possible to chain a `.catch()` on a rejected promise well after it's already been fulfilled.

Hmm… Is the sole purpose here to avoid the browser's default behaviour of emitting a console error message for an "unhandled rejection" (incorrectly/prematurely sometimes, though it doesn't know that). That default [rejection event behaviour](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises#promise_rejection_events) is changeable.

Reply:
My understanding is that you don't have control over unhandled rejections on a per-promise level. In general, I _do_ want unhandled rejections to result in error logs as most of the time they are coding errors, but not in this case. I'll update the message to indicate this is why the promises should not reject.

Reply:
Yeah, not easily on a per-promise level; you do always have to interact in some way with the global events/handlers.

Definitely want to surface unhandled rejections somehow for the reason that they're a programming error, yes, though I think the default global events/handlers are a very minimal, not very good example of that surfacing. As a example of an alternative, I'd consider the approach I used here as an improvement:

https://github.com/nextstrain/nextstrain.org/blob/01724a5d3669a936c1322b79b04289c26901952b/src/utils/scripts.js#L40-L76

This example is from Node.js and uses process exit as the trigger for reporting as-of-yet-unhandled rejections. In a browser context, you could use other triggers (likely more frequent), such as "finished loading" events specific to the app (e.g. a periodic barrier point by which all promises "should" be handled) or even something as simple as a regular interval (e.g. of a ~10s or something) so that promises have some space to get dealt.

Reply:
In this case these promises should be handled when Auspice renders the dataset they represent, which may never happen (e.g. if a narrative never renders the slide backed by this dataset). I think in every other case we attach `catch` clauses (or use try/catch + async) when the promise is set up, so the default behavior seems fine. At least, I can't remember seeing an "unhandled promise rejection" log which was not indicative of a bug (in the code modified here we were implicitly resolving to `undefined`, which had its own problems).